File: 3sem/4lab/Banks.Console/Handlers/RegisterClientHandler.cs
using Banks.Entities;
using Banks.Services;

namespace Banks.Console.Handlers;

public class RegisterClientHandler : Handler
{
    public override void Handle(string request, CentralBank centralBank)
    {
        ArgumentNullException.ThrowIfNull(request);
        ArgumentNullException.ThrowIfNull(centralBank);

        if (request.ToLower() == "register client")
        {
            System.Console.Write("Bank id: ");
            var bankId = Guid.Parse(System.Console.ReadLine() ?? throw new ArgumentNullException());

            System.Console.Write("Client name: ");
            var clientName = System.Console.ReadLine() ?? throw new ArgumentNullException();

            System.Console.Write("Client surname: ");
            var clientSurname = System.Console.ReadLine() ?? throw new ArgumentNullException();

            Bank bank = centralBank.GetBankById(bankId);
            var client = new Client(clientName, clientSurname);
            bank.AddClient(client);

            System.Console.WriteLine($"Client '{clientName} {clientSurname}' {client.Id} was created");
            return;
        }

        NextHandler?.Handle(request, centralBank);
    }
}

File: 4sem/1lab/src/main/java/BanksRemake/Presentation/Handlers/RegisterClientHandler.java
package BanksRemake.Presentation.Handlers;

import BanksRemake.Domain.Entities.Bank;
import BanksRemake.Domain.Entities.Client;
import BanksRemake.Domain.Exceptions.InvalidBankAccountOperationException;
import BanksRemake.Domain.Exceptions.InvalidBankOperationException;
import BanksRemake.Domain.Exceptions.InvalidDepositInterestRateSelectorException;
import BanksRemake.Domain.Exceptions.InvalidMoneyGapException;
import BanksRemake.Domain.Services.CentralBank;

import java.util.Scanner;
import java.util.UUID;

public class RegisterClientHandler extends HandlerBase {
    @Override
    public void handle(String request, CentralBank centralBank, Scanner scanner) throws InvalidBankOperationException, InvalidBankAccountOperationException, InvalidDepositInterestRateSelectorException, InvalidMoneyGapException {
        if (request.toLowerCase().equals("register client")) {
            System.out.print("Bank id: ");
            var bankId = UUID.fromString(scanner.nextLine());

            System.out.print("Client name: ");
            var clientName = scanner.nextLine();

            Bank bank = centralBank.getBankById(bankId);
            var client = new Client(clientName);
            bank.addClient(client);

            System.out.println(String.format("Client '%s' %s was created", client.getName(), client.getId()));
            return;
        }

        if (getNext() != null) {
            getNext().handle(request, centralBank, scanner);
        }
    }
}
